waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/MetaStoreMappingFactoryImpl.java

```diff
@@ -91,6 +97,30 @@ private CloseableThriftHiveMetastoreIface newUnreachableMetastoreClient(Abstract
             new UnreachableMetastoreClientInvocationHandler(metaStore.getName()));
   }
 
+  private MetaStoreFilterHook loadMetastoreFilterHook(AbstractMetaStore metaStore) {
+    HiveConf conf = new HiveConf();
+    String metaStoreFilterHook = metaStore.getHiveMetastoreFilterHook();
+    if (metaStoreFilterHook == null || metaStoreFilterHook.isEmpty()) {
+      return new DefaultMetaStoreFilterHookImpl(conf);
+    }
+    for (Map.Entry<String, String> property: waggleDanceConfiguration.getConfigurationProperties().entrySet()) {
+      conf.set(property.getKey(), property.getValue());
+    }
+    conf.set(HiveConf.ConfVars.METASTORE_FILTER_HOOK.varname, metaStoreFilterHook);
+//        "com.hotels.bdp.waggledance.mapping.model.AlluxioMetastoreFilter");
+    Class<? extends MetaStoreFilterHook> filterHookClass = conf.getClass(
+        HiveConf.ConfVars.METASTORE_FILTER_HOOK.varname, DefaultMetaStoreFilterHookImpl.class,
+        MetaStoreFilterHook.class);
+    String msg = "Unable to create instance of " + filterHookClass.getName() + ": ";
```

Comment:
You can move this down to line 120 no? (I understand this draft and stuff here likely to change and be tidied up).

Reply:
yes, moved it